src/targets/node/terminalNodeLauncher.ts

```diff
@@ -174,6 +175,35 @@ export class TerminalNodeLauncher extends NodeLauncherBase<ITerminalLaunchConfig
     });
   }
 
+  private async writeCommand(terminal: vscode.Terminal, command: string) {
+    // Wait for shell integration if we expect it to be available
+    const config = vscode.workspace.getConfiguration();
+    if (!terminal.shellIntegration && config.get(SHELL_INTEGRATION_SETTING) !== false) {
```

Comment:
Shell integration may activate or not regardless of this setting, the setting only determines whether we try to automatically inject it. It could fail or it could be manually installed in their start up scripts. So you shouldn't be checking this setting here.

Reply:
My thought was just to avoid the 1s race if I see the user has explicitly disabled it

Reply:
But then you're degrading the experience for users that have manually installed it, which are arguably those that are more engaged with the feature as they've gone and sought it out.